gnucash/gnome/business-urls.c

```diff
@@ -168,78 +168,77 @@ jobCB (const char *location, const char *label,
   {                                                                     \
     result->error_message =                                             \
       g_strdup_printf (_("No such owner entity: %s"), location);        \
-    return FALSE;                                                       \
+    show_report = FALSE;                                                \
   }
 
+/* parses a string "user=john&pass=smith&age=41" into a string-keyed
+   GHashTable. String must not contain non-ASCII chars. Duplicate keys
+   will be ignored. */
+static GHashTable *parse_parameters (const gchar *parms)
+{
+    GHashTable *rethash;
+    gchar *query, *p;
+
+    rethash = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+    query = g_strdup (parms);
+
+    while ((p = strsep (&query, "&\n")))
+    {
+        gchar *var = strtok (p, "=");
```

Comment:
Since there should be exactly one instance of '=' in the string this would be more efficient:
```C
gchar * val = strchr (p, '=');
*(val++) = '\0';
if (val && !g_has_table_contains (rethash, p))
   g_hash_table_insert (rethash, g_strdup (p), g_strdup (val));
```